app/models/playlist.rb

```diff
@@ -0,0 +1,50 @@
+# == Schema Information
+#
+# Table name: playlists
+#
+#  id            :bigint           not null, primary key
+#  description   :string
+#  name          :string           not null
+#  playlist_type :integer          not null
+#  private       :boolean          default(FALSE)
+#  created_at    :datetime         not null
+#  updated_at    :datetime         not null
+#  user_id       :bigint
+#
+class Playlist < ApplicationRecord
+  belongs_to :user, optional: true
```

Comment:
Always storing the creator and keeping the three levels of access (private/personal/public) in an enum makes more sense to me, and it feels like it would be harder to mess up access with that system.

Reply:
I've implemented this, i agree that it makes things a lot simpler. I went for `secret` rather than `private` since an enum can't define the method private (I collides with a method from ActiveRecord).

One extra point here: The user_id now only gets set add creation. This means that if user X makes playlist and then user Y makes it personal or secret, it still belongs to user X. The other option would have been to always set the user_id to the last user who edited the playlist.
I don't think this is a real world problem (since shared playlist already require the trust that another user would not just delete all the items), but wanted to point it out as an edge case

Reply:
Is access an attribute we want to be able to change after creation? (I don't immediately see a use-case.)

Reply:
I'd say so, mostly because someone might change their mind on something being personal/secret (or just create a playlist a select the wrong access by accident)